easyfinance.client/src/app/features/income/list-incomes/list-incomes.component.html

```diff
@@ -43,46 +43,44 @@ <h5 class="mb-1 name">{{ income.name }}</h5>
           </div>
           } @else {
           <div>
-            <form class="d-flex justify-content-between align-items-center my-1" [formGroup]="incomeForm" (ngSubmit)="save()">
-              <div class="form-floating">
-                <input type="text" class="form-control" id="name" placeholder="Supermarket" formControlName="name" [class.is-invalid]="name?.invalid && name?.touched">
-                <label for="name">Income Name</label>
-                @if(name?.invalid && name?.touched){
-                <div>
-                  @if(name?.errors?.['required']){
-                  <div class="text-danger">Income Name is required.</div>
-                  }
-                </div>
-                }
-              </div>
-              <div class="form-floating">
-                <input type="text" class="form-control" id="date" formControlName="date" [class.is-invalid]="date?.invalid && date?.touched">
-                <label for="date">Date</label>
-                @if(date?.invalid && date?.touched){
-                <div>
-                  @if(date?.errors?.['required']){
-                  <div class="text-danger">Income Date is required.</div>
+            <form [formGroup]="incomeForm" (ngSubmit)="save()">
+
+              <div class="row">
+                <mat-form-field appearance="fill" class="col-md-6">
+                  <mat-label>Income Name</mat-label>
+                  <input matInput id="name" placeholder="Personal" formControlName="name"
+                         [class.is-invalid]="name?.invalid && name?.touched">
+                  @if ((name?.invalid && name?.touched) || httpErrors){
+                    <mat-error *ngFor="let error of getFormFieldErrors('name')">{{ error }}</mat-error>
                   }
-                  @if(date?.errors?.['pattern']){
-                  <div class="text-danger">Invalid Date format. (yyyy-MM-dd).</div>
+                </mat-form-field>
+
+                <mat-form-field appearance="fill" class="col-md-6">
+                  <mat-label>Date</mat-label>
+                  <input matInput [matDatepicker]="picker" formControlName="date">
```

Comment:
```suggestion
                  <input matInput [matDatepicker]="picker" id="date" formControlName="date">
```
You can add the id date to the input or you can change the test to find by formControlName, what you decided I will aprove. Just to pass the test.

Thank you again, good code I liked a lot.

Reply:
Sure, I will consider that.

I'm so glad to know that.
Thanks for your encouragement.